eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Utxo.hs

```diff
@@ -516,7 +516,7 @@ validateValueNotConservedUTxO ::
   PParams era ->
   UTxO era ->
   CertState era ->
-  TxBody era ->
+  TxBody t era ->
```

Comment:
Isn't this something that only applies to TopLevel? The fact that consumed = produced?  Shouldn't it be `TxBody TopTx era`